src/client/app/components/ReadingsPerDaySelectComponent.tsx

```diff
@@ -2,58 +2,45 @@
-import { updateThreeDReadingInterval } from '../actions/graph';
-import { ByMeterOrGroup, MeterOrGroup } from '../types/redux/graph';
-import { roundTimeIntervalForFetch } from '../utils/dateRangeCompatibility';
-import * as moment from 'moment';
 
 /**
  * A component which allows users to select number of reading per day for the graphic
  * @returns A Select menu with Readings per day options.
  */
 export default function ReadingsPerDaySelect() {
-	const dispatch: Dispatch = useDispatch();
-	const graphState = useSelector((state: State) => state.graph);
-	// Level of detail along the xAxis / Readings per day
-	const readingInterval = useSelector((state: State) => state.graph.threeD.readingInterval);
-	// This makes sure that the time between graphic values is not less than the time between readings.
-	const actualReadingInterval = useSelector((state: State) => {
-		const threeDState = state.graph.threeD;
-		const meterOrGroupID = threeDState.meterOrGroupID;
-		// meterOrGroup determines whether to get readings from state .byMeterID or .byGroupID
-		const byMeterOrGroup = threeDState.meterOrGroup === MeterOrGroup.meters ? ByMeterOrGroup.meters : ByMeterOrGroup.groups;
-		// 3D requires intervals to be rounded to a full day.
-		const timeInterval = roundTimeIntervalForFetch(state.graph.timeInterval).toString();
-		const unitID = state.graph.selectedUnit;
+	const dispatch = useAppDispatch();
+	const graphState = useAppSelector(selectGraphState);
+	const readingInterval = useAppSelector(selectThreeDReadingInterval);
+	const { args, shouldSkipQuery } = useAppSelector(selectThreeDQueryArgs);
 
-		// If meter or group not selected return null data, else return data if any.
-		const data = !meterOrGroupID ? null : state.readings.threeD[byMeterOrGroup][meterOrGroupID]?.[timeInterval]?.[unitID]?.[readingInterval]?.readings;
+	const { data, isFetching } = readingsApi.endpoints.threeD.useQuery(args, { skip: shouldSkipQuery });
 
-		if (data && data.zData.length) {
-			// Special Case:  When no compatible data available, data returned is from api is -999
-			if (data.zData[0][0] && data.zData[0][0] < 0) {
-				return ReadingInterval.Incompatible;
-			}
-
-			// Calculate the actual time interval based on the xLabel values
+	let actualReadingInterval = ReadingInterval.Hourly
+	if (data && data.zData.length) {
+		// Special Case:  When no compatible data available, data returned is from api is -999
+		if (data.zData[0][0] && data.zData[0][0] < 0) {
+			actualReadingInterval = ReadingInterval.Incompatible;
+		} else {
 			const startTS = moment.utc(data.xData[0].startTimestamp);
 			const endTS = moment.utc(data.xData[0].endTimestamp);
 			// This should be the number of hours between readings.
-			const actualReadingInterval = endTS.diff(startTS) / 3600000;
-			return actualReadingInterval
+			actualReadingInterval = endTS.diff(startTS) / 3600000;
 		}
 
-		// Return normal interval
-		return readingInterval;
-	})
+	}
+
+	// Return normal interval
```

Comment:
First, could this commented out code be removed? Also, I'm unclear on why the new code returns the same value as the old code in this case. Finally, should there be a return above and not be implicit?

Reply:
Commented out code, removed.

Reply:
What do you mean by old code. vs new code. I'm unsure there's a difference between the two iterations in this context. Could you expand on this?